CoreTimelineView/src/au/gov/asd/tac/constellation/views/timeline/TimelineTopComponent.java

```diff
@@ -213,7 +208,70 @@ public void setExtents(final double lowerTimeExtent, final double upperTimeExten
         }
     }
 
-    // <editor-fold defaultstate="collapsed" desc="Timeline and Histogram Extents">
+    private boolean videoParamsSet = false;
+    private int videoPos = -1;
+    private double originalLowerTime;
+    private double originalUpperTime;
+
+    /**
+     * Plays a video of the current selection
+     *
+     */
+    public void playVideo() throws InterruptedException {
+        resetVideo();
+        videoPos = -1;
+        originalLowerTime = getTimelineLowerTimeExtent();
+        originalUpperTime = getTimelineUpperTimeExtent();
+        videoParamsSet = true;
+
+        while (videoPos < 100 && videoParamsSet) {
+            double newLowerTime = (originalUpperTime - originalLowerTime) / 101 * videoPos + originalLowerTime;
+            double newUpperTime = (originalUpperTime - originalLowerTime) / 101 * (videoPos + 1) + originalLowerTime;
+            setExtents(newLowerTime, newUpperTime);
+            videoPos++;
+            Thread.sleep(100, 0);
```

Comment:
Curious as to the need for the sleep calls?
Is this so the current frame is viewed for 100ms ?